AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/MutationSync/OutgoingMutationQueue/SyncMutationToCloudOperation.swift

```diff
@@ -179,6 +181,17 @@ class SyncMutationToCloudOperation: Operation {
         }
     }
 
+    private func resolveReachabilityPublisher(request: GraphQLRequest<MutationSync<AnyModel>>) {
+        if networkReachabilityPublisher == nil {
+            guard let apiName = request.apiName else {
+                return
+            }
+            if let reachability = api as? APICategoryReachabilityBehavior {
+                networkReachabilityPublisher = reachability.reachability(apiName: apiName)
+            }
```

Comment:
Tim mentioned that we could hide behind an interface, it looking through the code, it became unclear on how to accomplish this.  Any suggestions on how to expose this functionality to other categories would be appreciated.

Reply:
i'm not entirely sure, is it possible to move `APICategoryReachabilityBehavior` over to APIPlugin? APIPlugin can still implement the behavior but developer will not be able to call it from Amplify.API.<method>. they would also have to check that `api as? APIReachabilityBehavior` to get the methods